src/taskEvents/script.js

```diff
@@ -7,23 +7,49 @@ import {
   addLoader,
   addErrorMessage,
   removeLoader,
+  getUserSkills,
+  getTagLevelOptions,
+  addSkillToUser,
 } from './utils.js';
 
+import roleElement from './roleElement.js';
+
 import { createEventCard } from './logCard.js';
 
+let tagLevelOptions;
 const container = document.getElementById('task-events-container');
 const modal = document.getElementById('modal');
 const overlay = document.querySelector('.overlay');
+export const modalOverlay = createElement({
+  type: 'div',
+  attributes: {
+    class: 'modal-overlay',
+  },
+});
+const modalOverlayLoader = createElement({
+  type: 'div',
+  attributes: {
+    class: 'dot-flashing-loader',
+  },
+});
+modal.appendChild(modalOverlay);
+modalOverlay.appendChild(modalOverlayLoader);
 
 const closeBtn = document.getElementById('close-btn');
 closeBtn.addEventListener('click', closeModal);
 const genericModalCloseBtn = document.getElementById('close-generic-modal');
 genericModalCloseBtn.addEventListener('click', closeGenericModal);
 
+addEventListener('load', async (event) => {
+  tagLevelOptions = await getTagLevelOptions();
+});
+
 async function createProfileModal(username) {
   try {
     overlay.classList.remove('hidden');
     const { user } = await getUserData(username);
+    const { skills } = await getUserSkills(user.id);
+    const rolesDiv = createRolesDiv(skills, user.id);
```

Comment:
Passing `skills` to `createRolesDiv()` and storing it as `roles` inside the function seems confusing. Please check

Reply:
yeah will make the change to avoid confusion

Reply:
done